src/lib/three-d-secure.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+var fs = require('fs');
+var classlist = require('./classlist');
+var threeDSecure = require('braintree-web/three-d-secure');
+var Promise = require('./promise');
+
+function ThreeDSecure(client, merchantConfiguration, cardVerificationString) {
+  this._client = client;
+  this._config = merchantConfiguration;
+  this._modal = this._setupModal(cardVerificationString);
+}
+
+ThreeDSecure.prototype.initialize = function () {
+  var self = this;
+
+  return threeDSecure.create({
+    client: this._client
+  }).then(function (instance) {
+    self._instance = instance;
+  });
+};
+
+ThreeDSecure.prototype.verify = function (nonce) {
+  var self = this;
+
+  this._revealModal();
+
+  return Promise.all([
```

Comment:
this is a bit tangled to me but not sure how to make it easier to read off the top of my head